build-system/tasks/e2e/functional-test-controller.js

```diff
@@ -13,6 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+// @ts-nocheck
```

Comment:
Is this intentional? If yes, maybe a better option is to exclude this file via `build-system/tsconfig.json`?

Reply:
Unfortunately that won't work because the file is imported by other files which are in the `included` list. (It's not very intuitive behavior) https://github.com/microsoft/TypeScript/issues/36017 

That being said in the time since this PR was written and now the number of issues in he build-system directory has ballooned again so I think I'll just remove this for now and try to fix the root issue in another PR.